src/torchcodec/decoders/_core/VideoDecoder.h

```diff
@@ -225,7 +225,8 @@ class VideoDecoder {
   // seconds=5.999, etc.
   DecodedOutput getFramePlayedAtTimestampNoDemux(double seconds);
 
-  DecodedOutput getFrameAtIndex(
+  DecodedOutput getFrameAtIndex(int streamIndex, int64_t frameIndex);
+  DecodedOutput getFrameAtIndexInternal(
```

Comment:
I had to move this one back to public unfortunately, because we must now rely on this one for our C++ tests. Eventually once we accept pre-allocated tensors in all high-level entrypoints, we'll be able to move it back to private.

Reply:
We could also use the [friend](https://en.cppreference.com/w/cpp/language/friend) declaration, but I'd prefer not to even mention the testing class in the implementation. I think this is fine.